### ISSUE TITLE
SourceMap loader complains "Missing source files" in the new v8 build

### ISSUE THREAD

**Issue by okhomenko** [maslianok/react-resize-detector#228]
```
WARNING in ./node_modules/react-resize-detector/build/index.esm.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/src/main/webpack/node_modules/react-resize-detector/src/ResizeDetector.tsx' file: Error: ENOENT: no such file or directory, open '/src/main/webpack/node_modules/react-resize-detector/src/ResizeDetector.tsx'
Error: Failed to parse source map from '/src/main/webpack/node_modules/react-resize-detector/src/ResizeDetector.tsx' file: Error: ENOENT: no such file or directory, open '/src/main/webpack/node_modules/react-resize-detector/src/ResizeDetector.tsx'
    at fetchFromFilesystem (/src/main/webpack/node_modules/source-map-loader/dist/utils.js:140:11)
    at async fetchFromURL (/src/main/webpack/node_modules/source-map-loader/dist/utils.js:254:9)
    at async /src/main/webpack/node_modules/source-map-loader/dist/index.js:105:11
```

**Comment by snelsi** [comment]
Please, provide some more info about your project's setup:
- Do you use custom `Webpack` / `source-map-loader` config?
- Do you see this error when installing node modules? Or during dev / build commands?

I tried to google the error message and it seems like it's just a false-positive warning by the latest CRA 5 / Webpack 5 release:
https://github.com/mebjas/html5-qrcode/issues/396#issuecomment-1067145379
https://github.com/colinhacks/zod/issues/828#issuecomment-1005006072

You can hide it temporarly by setting `GENERATE_SOURCEMAP` variable to `false` in your app.

Based on links below, adding `src` to the `files` field in the `package.json` might fix the warning on our side as well:
https://github.com/mswjs/interceptors/pull/204
https://github.com/mswjs/msw/issues/1030#issuecomment-1024251944
https://github.com/colinhacks/zod/pull/864

Or we can just remove `.map` files from the build.

**Comment by snelsi** [comment]
@maslianok I'm more leaning toward removing `.map` files from the build, based on [this discussion](https://github.com/colinhacks/zod/pull/864#issuecomment-1056079209). What do you think?

**Comment by maslianok** [comment]
We haven't published library sources to npm before. Does it mean that the library's source maps didn't work at all? 😮 

Well, looks like it's a good idea to remove files that are useless 😄

**Comment by maslianok** [comment]
Removed in v8.0.1
PR #229

**Comment by okhomenko** [comment]
Source maps worked in v7. For each of the src/* files you had a section in `sourcesContent`
<img width="1047" alt="Screenshot 2023-01-18 at 3 26 01 PM" src="https://user-images.githubusercontent.com/206691/213308844-ca5d5344-2a12-4fd6-aaed-847c6b64c5fe.png">

**Comment by snelsi** [comment]
@okhomenko You are right, sourcesContent is `null` is the new version for some reason.

Looks like to fix this is just a matter of adding `inlineSources: true` to the `tsconfig:`
https://github.com/rollup/plugins/issues/260

![image](https://user-images.githubusercontent.com/15171036/213374921-8c6038aa-1b21-4e03-98b3-e19aad1f2292.png)

**Comment by maslianok** [comment]
Do we increase production build size by adding sourcemaps/sources to the module?

**Comment by maslianok** [comment]
I did some testing and it seems my understanding of sourcemaps was correct. 
Sourcemaps are included ONLY in development builds. When you run "yarn build" in CRA it creates an optimized prod build without sourcemaps. 

Firs run: current v8.0.1 build without sourcemaps
Second run: with `sourcemap: true` & `inlineSources: true`
 
<img width="495" alt="2023-01-19 о 09 41 37" src="https://user-images.githubusercontent.com/3485490/213396436-029fc769-c64a-4c07-8d66-ed665f0b2da3.png">


I suppose other modern libraries also follow this rule.

Can you confirm or deny this?

**Comment by snelsi** [comment]
Can confirm that the only difference between the build with `inlineSources: false` and `inlineSources: true` are `sourcesContent` field inside `.map files`. 

Either to include or exclude the source map files from the build decides the user, really. 
By default they are excluded in the majority of the prod builds.

Looks good to me to merge in the master 👍:
https://github.com/maslianok/react-resize-detector/pull/230

**Comment by maslianok** [comment]
Merged and published v8.0.2.

Thank you, @snelsi 

@okhomenko please confirm that it works as you expected
